internal/impl/azure/shared/service.go

```diff
@@ -121,3 +123,48 @@ func parseConnectionString(input string) (storageAccount, storageAccessKey strin
 	}
 	return accountName, accountKey, nil
 }
+
+// GetStorageClient creates the storage client for the provided credentials
+// TODO: Use a proper config struct for the parameters to avoid mixing them up
+func GetStorageClient(storageConnectionString, storageAccount, storageAccessKey, storageSASToken string) (*azblob.Client, error) {
```

Comment:
I would rename this to GetBlobStorageClient to be more explicit because Azure storage comprises blob, file and queue.